Makefile.docker

```diff
@@ -2,6 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 UTC_DATE=$(shell date -u "+%Y-%m-%d")
+REPOSITORY ?= quay.io
```

Comment:
while this one is making sense,I can't find anywhere where REPOSITORY is passed as docker.io, will cilium publish images to docker.io anymore ?

Reply:
@sayboras the images aren't "pushed", both docker and quay have rules to automatically build images once a tag is released. The GH actions that we have are currently doing "push" to both `quay.io` and `docker.io` but to the `*-dev` repositories.